packages/lib-classifier/src/store/WorkflowStepStore.js

```diff
@@ -150,41 +171,81 @@ const WorkflowStepStore = types
 
     function convertWorkflowToUseSteps (workflow) {
       const taskKeys = Object.keys(workflow.tasks)
-      const { first_task } = workflow
 
-      function getStepTasksFromCombo (task) {
-        task.tasks.forEach(function (taskKey) {
-          taskKeys.splice(taskKeys.indexOf(taskKey), 1)
-        })
-        return task.tasks
+      function getTaskKeysIncludedInComboTasks (tasks) {
+        let taskKeys
+        const comboTasks = Object.values(tasks).filter(task => task?.type === 'combo')
+        taskKeys = comboTasks.map(combo => combo.tasks)
+        return taskKeys.flat()
       }
+      const taskKeysIncludedInComboTasks = getTaskKeysIncludedInComboTasks(workflow.tasks)
+      const taskKeysToConvertToSteps = difference(taskKeys, taskKeysIncludedInComboTasks)
+
+      const { first_task } = workflow
+      const firstTask = workflow.tasks[first_task]
 
       if (first_task) {
         let firstStep = {
+          next: firstTask.next, // temporarily set next to task key, convert to step key once steps created
           stepKey: 'S0',
           taskKeys: [first_task]
         }
 
-        if (workflow.tasks[first_task].type === 'combo') {
-          const combo = workflow.tasks[first_task]
-          firstStep.taskKeys = getStepTasksFromCombo(combo)
+        if (firstTask.type === 'combo') {
+          const combo = firstTask
+          firstStep.taskKeys = combo.tasks
```

Comment:
This could be shorten to:
```suggestion
          firstStep.taskKeys = firstTask.tasks
```